agent/consul/acl_endpoint_test.go

```diff
@@ -1026,31 +1468,78 @@ func TestACLEndpoint_TokenList(t *testing.T) {
 
 	testrpc.WaitForLeader(t, s1.RPC, "dc1")
 
-	t1, err := upsertTestToken(codec, "root", "dc1")
+	clock := newStoppedClock(time.Now())
+	expTime := clock.Now().Add(2 * time.Minute)
+
+	acl := s1.GetEndpoint("ACL").(*ACL)
+	acl.clock = clock
+
+	t1, err := upsertTestToken(codec, "root", "dc1", nil)
 	require.NoError(t, err)
 
-	t2, err := upsertTestToken(codec, "root", "dc1")
+	t2, err := upsertTestToken(codec, "root", "dc1", nil)
 	require.NoError(t, err)
 
-	acl := ACL{srv: s1}
+	t3, err := upsertTestToken(codec, "root", "dc1", func(token *structs.ACLToken) {
+		token.ExpirationTime = expTime
+	})
+	require.NoError(t, err)
 
-	req := structs.ACLTokenListRequest{
-		Datacenter:   "dc1",
-		QueryOptions: structs.QueryOptions{Token: "root"},
-	}
+	masterTokenAccessorID, err := retrieveTestTokenAccessorForSecret(codec, "root", "dc1", "root")
+	require.NoError(t, err)
 
-	resp := structs.ACLTokenListResponse{}
+	t.Run("normal", func(t *testing.T) {
+		req := structs.ACLTokenListRequest{
+			Datacenter:   "dc1",
+			QueryOptions: structs.QueryOptions{Token: "root"},
+		}
 
-	err = acl.TokenList(&req, &resp)
-	require.NoError(t, err)
+		resp := structs.ACLTokenListResponse{}
 
-	tokens := []string{t1.AccessorID, t2.AccessorID}
-	var retrievedTokens []string
+		err = acl.TokenList(&req, &resp)
+		require.NoError(t, err)
 
-	for _, v := range resp.Tokens {
-		retrievedTokens = append(retrievedTokens, v.AccessorID)
-	}
-	require.Subset(t, retrievedTokens, tokens)
```

Comment:
I removed use of `require.Subset` in this whole file because it has the potential (as I encountered legitimately) to mask bugs in returning data the function should not.